### PR TITLE
Add ophys GIN

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

Adding coverage to all `ophys` interfaces, revealing some bugs that have changed over the months with various API changes.

## How to test the behavior?
Thanks to @Saksham20 fixing the GIN for this data, the caching CI now checks for updates every night along with the ecephys data.

I've then added two simple `Imaging` and `Segmentation` interfaces to the `test_gin.py` so @cechava can begin investigating a fix. Will add all the others we currently support later today.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@bendichter I finally got the `ophys` tests working and the interfaces debugged. The metadata had never been validated at this level before, which revealed some issues. While those issues have, here, been temporarily resolved in a non-elegant fashion, a future PR will do a better job cleaning up underlying issues. But for now, I do think we should try to get this into main as quickly as possible since none of our current `ophys` interfaces actually work without these fixes.

File: nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py [hunk 1]
```diff
@@ -53,6 +55,18 @@ def get_metadata(self):
...
+        # fix troublesome data types
+        if "TwoPhotonSeries" in metadata["Ophys"].keys():
+            for i in range(len(metadata["Ophys"]["TwoPhotonSeries"])):
+                if "dimension" in metadata["Ophys"]["TwoPhotonSeries"][i].keys():
+                    metadata["Ophys"]["TwoPhotonSeries"][i]["dimension"] = list(
+                        metadata["Ophys"]["TwoPhotonSeries"][i]["dimension"]
+                    )
+                if "rate" in metadata["Ophys"]["TwoPhotonSeries"][i].keys():
+                    metadata["Ophys"]["TwoPhotonSeries"][i]["rate"] = float(
+                        metadata["Ophys"]["TwoPhotonSeries"][i]["rate"]
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py:68]:
```suggestion
            for two_photon_series in metadata["Ophys"]["TwoPhotonSeries"]:
                if "dimension" in two_photon_series:
                    two_photon_series["dimension"] = list(
                        two_photon_series["dimension"]
                    )
                if "rate" in two_photon_series:
                    two_photon_series["rate"] = float(
                        two_photon_series["rate"]
```

File: nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py [hunk 2]
```diff
@@ -53,6 +55,18 @@ def get_metadata(self):
         metadata = super().get_metadata()
         metadata.update(re.NwbImagingExtractor.get_nwb_metadata(self.imaging_extractor))
         _ = metadata.pop("NWBFile")
+
+        # fix troublesome data types
+        if "TwoPhotonSeries" in metadata["Ophys"].keys():
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ophys/baseimagingextractorinterface.py:60]:
```suggestion
        if "TwoPhotonSeries" in metadata["Ophys"]:
```

**Comment by bendichter** [comment]
made a few syntax suggestions.`str_ in dict_.keys()` should be `str_ in dict_`

**Comment by CodyCBakerPhD** [comment]
Good catches; those were from Cesar's first changes that I forgot to double-check and review

**Review by bendichter — APPROVED** [review]

**Pull request merged**